### PR TITLE
[MB-10959] Upgrade eslint airbnb to v19 and fix linting errors

### PR DESCRIPTION
**Pull request by duncan-truss**
## Summary

This upgrades the `eslint-config-airbnb` library to version 19.  Dependabot was failing with an invalid configuration for a new rule `react/function-component-definition` that was introduced.  The real issue is that we are still on v4 of react-scripts and weren't getting a newer version of the `eslint-plugin-react` library that was required, so I added that with the required minimum [until we can upgrade react-scripts properly (currently blocked)](https://dp3.atlassian.net/browse/MB-10955).

I didn't choose a single configuration for `react/function-component-definition` and instead set it to allow for multiple styles.  It may have introduced a ton of extra changes to this PR if I say settled on `arrow-function` as the way to go for everything in the codebase.  I thought it was better to keep this focused on the version changes and #prac-frontend can follow up on a recommendation.  This rule does tend to be auto fixable though so it may just be a matter of running `yarn lint --fix`

https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/function-component-definition.md

These were the main linter errors that I had to fix:

Screen Shot 2022-01-31 at 9 39 12 AM

##### Terminal 1

Run ESLint on the project files

```sh
yarn install
yarn lint
```

Verify that we can run the client server without errors
```sh
make client_run
```

Verify that we can run storybook without errors
```sh
make storybook
```

### Additional steps

1. Access the
2. Login as a
3.

## Verification Steps for Author

These are to be checked by the author.

- [ ] Tested in the Experimental environment (for changes to containers, app startup, or connection to data stores)
- [ ] Request review from a member of a different team.
- [ ] Have the Jira acceptance criteria been met for this change?

## Verification Steps for Reviewers

These are to be checked by a reviewer.

### Frontend

- [ ] User facing changes have been reviewed by design.
- [ ] There are no aXe warnings for UI.
- [ ] This works in [Supported Browsers and their phone views](https://github.com/transcom/mymove/tree/master/docs/adr/0016-Browser-Support.md) (Chrome, Firefox, IE, Edge).
- [ ] There are no new console errors in the browser devtools
- [ ] There are no new console errors in the test output
- [ ] If this PR adds a new component to Storybook, it ensures the component is fully responsive, OR if it is intentionally not, a wrapping div using the `officeApp` class or custom `min-width` styling is used to hide any states the would not be visible to the user.

### Backend

- [ ] Code follows the guidelines for [Logging](https://transcom.github.io/mymove-docs/docs/dev/contributing/backend/Backend-Programming-Guide#logging)
- [ ] The requirements listed in [Querying the Database Safely](https://transcom.github.io/mymove-docs/docs/dev/contributing/backend/Backend-Programming-Guide/#querying-the-database-safely) have been satisfied.

#### Any new migrations/schema changes:

- [ ] Follows our guidelines for [Zero-Downtime Deploys](https://transcom.github.io/mymove-docs/docs/dev/contributing/database/Database-Migrations#zero-downtime-migrations)
- [ ] Have been communicated to #g-database
- [ ] Secure migrations have been tested following the instructions in our [docs](https://transcom.github.io/mymove-docs/docs/dev/contributing/database/Database-Migrations#secure-migrations)

### CODE REVIEW

**Review by felipe-lee — APPROVED** [review]
> This upgrades the eslint-config-airbnb library to version 19. Dependabot was failing with an invalid configuration for a new rule react/function-component-definition that was introduced. The real issue is that we are still on v4 of react-scripts and weren't getting a newer version of the eslint-plugin-react library that was required, so I added that with the required minimum until we can upgrade react-scripts properly (currently blocked).

I added a comment to https://dp3.atlassian.net/browse/MB-10955 to remove the resolution once `react-scripts` is updated.

> I didn't choose a single configuration for react/function-component-definition and instead set it to allow for multiple styles. It may have introduced a ton of extra changes to this PR if I say settled on arrow-function as the way to go for everything in the codebase. I thought it was better to keep this focused on the version changes and #prac-frontend can follow up on a recommendation. This rule does tend to be auto fixable though so it may just be a matter of running yarn lint --fix

Yeah I agree that for now the version update is good to focus on.

The changes lgtm. I tested the customer and office apps (very lightly) as well as storybook and all seem fine still. I just had a question, but it's likely more for me to learn than anything wrong haha.

File: src/components/Office/ServiceItemCalculations/helpers.js [hunk 1]
```diff
@@ -734,6 +734,6 @@ const makeCalculations = (itemCode, totalAmount, params, mtoParams) => {
       break;
   }
   return result;
-};
+}
 
-export { makeCalculations as default, makeCalculations };
+export { makeCalculations };
```
[the remark this answers is not in the document]
Comment (duncan-truss) [reply src/components/Office/ServiceItemCalculations/helpers.js:739]: So for this specific instance we have another linter rule `import/prefer-default-export`, which says that when there is only a single export from a file that it should use the default one over the named one.  I could get rid of the named one but the only place it's imported in ServiceItemCalculations.jsx it uses the `import { makeCalculations } from '.helpers.js'` syntax.  If I got rid of the named export then I could change it to `import makeCalculations from '.helpers.js'`.  Those are our choices, it really depends if we plan to add more helpers to this file over time, then the next person is just going to have to add it back.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/export

**Review by hannaliebl — APPROVED** [review]
LGTM, thanks for providing the image of the linter errors, it was helpful to follow along.

File: package.json [hunk 2]
```diff
@@ -6,7 +6,7 @@
     "@fortawesome/react-fontawesome": "^0.1.16",
     "@material-ui/core": "^4.12.3",
     "@material-ui/icons": "^4.11.2",
-    "@trussworks/react-file-viewer": "https://github.com/trussworks/react-file-viewer",
+    "@trussworks/react-file-viewer": "git+https://github.com/trussworks/react-file-viewer",
```
[the remark this answers is not in the document]
Comment (duncan-truss) [reply package.json:9]: Yea sorry that was a bit of a holdover from the previous security vulnerability PR and I'd try to run `npm list` and get an invalid error for that syntax.

**Review by gidjin — APPROVED** [review]
Reviewed the code changes. LGTM thank you for applying the lint fixes!

**Review by dinabeigelman — APPROVED** [review]
👍

**Pull request merged**